plugins/module_utils/sops.py

```diff
@@ -41,6 +46,69 @@
 }
 
 
+def _create_single_arg(argument_name):
+    def f(value, arguments, env):
+        arguments.extend([argument_name, to_native(value)])
+
+    return f
+
+
+def _create_comma_separated(argument_name):
+    def f(value, arguments, env):
+        arguments.extend([argument_name, ','.join([to_native(v) for v in value])])
+
+    return f
+
+
+def _create_repeated(argument_name):
+    def f(value, arguments, env):
+        for v in value:
+            arguments.extend([argument_name, to_native(v)])
+
+    return f
+
+
+def _create_boolean(argument_name):
+    def f(value, arguments, env):
+        if value:
+            arguments.append(argument_name)
+
+    return f
+
+
+def _create_env_variable(argument_name):
+    def f(value, arguments, env):
+        env[argument_name] = value
+
+    return f
+
+
+GENERAL_OPTIONS = {
+    'aws_profile': _create_single_arg('--aws-profile'),
+    'aws_access_key_id': _create_env_variable('AWS_ACCESS_KEY_ID'),
+    'aws_secret_access_key': _create_env_variable('AWS_SECRET_ACCESS_KEY'),
+    'aws_session_token': _create_env_variable('AWS_SESSION_TOKEN'),
+    'config_path': _create_single_arg('--config'),
+    'enable_local_keyservice': _create_boolean('--enable-local-keyservice'),
+    'keyservice': _create_repeated('--keyservice'),
+}
+
+
+ENCRYPT_OPTIONS = {
```

Comment:
I was wondering why dividing the options in 2 different sets. Even if they do apply only to encryption phase, would not be simpler to have them available as in the standard sops cli?

Reply:
This reduces the number of options the plugins/modules that do not encrypt have (and thus makes their docs less confusing).